src/foam/dao/EasyDAO.js

```diff
@@ -111,16 +115,25 @@ foam.CLASS({
         return this.nSpec && this.nSpec.name || (this.of && this.of.id);
       },
       javaFactory: `
-      if ( getNSpec() != null ) return getNSpec().getName();
-      if ( getOf()    != null ) return getOf().getId();
-      return "";
+      NSpec nspec = getNSpec();
+      if ( nspec != null ) return nspec.getName();
+      Loggers.logger(getX(), this).warning("NSpec not found");
+      if ( getOf() != null ) {
+        String id = getOf().getId();
+        String name = id.substring(id.lastIndexOf('.') + 1);
+        name += "DAO";
+        return foam.util.StringUtil.daoize(name);
+      } 
+      Loggers.logger(getX(), this).warning("Of not found");
+      return "EasyDAO: DAO not found";
      `
     },
     {
       name: 'nSpec',
       class: 'FObjectProperty',
       type: 'foam.nanos.boot.NSpec',
-      javaPostSet: 'setName(val.getName());'
+      javaFactory: 'return getX().get(NSpec.class);',
```

Comment:
What is this for?

Reply:
A few of the 'Aware' DAOs make use of the nspec name in their own setup.  The NSpecFactory places the NSpec in the calling/construction context.  EasyDAO is NSpecAware but this property is not always set before getName is called, so when not set, check the context for it.